src/util/processPool.ml

```diff
@@ -0,0 +1,44 @@
+type task = {
+  command: string;
+  cwd: string option;
+}
+
+let run ~jobs ?(terminated=fun _ _ -> ()) tasks =
+  let procs = Hashtbl.create jobs in
+  let rec run tasks =
+    match tasks with
+    | task :: tasks when Hashtbl.length procs < jobs ->
+      let old_cwd = Sys.getcwd () in
+      let proc =
+        match task.cwd with
+        | Some cwd ->
+          Fun.protect ~finally:(fun () ->
+              Sys.chdir old_cwd
+            ) (fun () ->
+              Sys.chdir cwd;
+              Unix.open_process task.command
+            )
+        | None ->
+          Unix.open_process task.command
+      in
+      let pid = Unix.process_pid proc in
+      Hashtbl.replace procs pid (task, proc);
+      run tasks
+    | [] when Hashtbl.length procs = 0 ->
+      ()
+    | _ ->
+      let (pid, status) = Unix.wait () in
```

Comment:
Maybe a quick comment about what `Unix.wait ()` does (wait for a child process to terminate) would be helpful to understand what's going on here without familiarity with the `Unix` module. The rest of the used functions is quite straight-forward.